src/client/thirdParty/Inkeep.tsx

```diff
@@ -26,17 +33,39 @@ type InkeepSharedSettings = {
   modalSettings: InkeepModalSettings;
 };
 
-const useInkeepSettings = (): InkeepSharedSettings => {
+export const createCustomAnalyticsCallback = (
```

Comment:
Is this indirection/callback factory necessary? Why not just declare the callback?

Reply:
It is needed as this is a callback defined by amplitude here:

    logEventCallback: customAnalyticsCallback,

And I now need to pass a property (visitorId). So I need this callback factory.

Reply:
No, I think you can simplify this :) 

1. Call `useVisitorData` to get the visitor ID inside `useInkeepSettings`,
2. Create `logEventCallback` inside `useInkeepSettings`